In [2]:
import pandas as pd
import numpy as np
from sklearn import linear_model
from sklearn.metrics import r2_score
from sklearn.metrics import mean_squared_error
from sklearn.model_selection import cross_val_score

In [3]:
#final = pd.read_csv("/Users/mdbarshay/Desktop/401/Project 3/final.csv", index_col = 0)
final =pd.read_csv("../../data/final.csv", index_col = 0)

In [4]:
X = final.iloc[:, :50]
y = final.iloc[:,50]

In [5]:
X = (X-X.mean())/X.std()
y = y / 100000

In [6]:
lm = linear_model.LinearRegression()
res = cross_val_score(lm, X, y, scoring = "neg_mean_squared_error", cv = 5)
res2 = cross_val_score(lm, X, y, scoring = "neg_root_mean_squared_error", cv = 5)
res3 = cross_val_score(lm, X, y, scoring = "r2", cv = 5)

The average mean squared error for 5-fold cross validated linear regression is:

In [7]:
-np.mean(res)

2817.7455042857828

The average root mean squared error for 5-fold cross validated linear regression is:

In [8]:
-np.mean(res2)

48.17603766774926

The average R^2 for 5-fold cross validated linear regression is:

In [9]:
-np.mean(res3)

0.3387162242076388

This means that on average our predictions for median house price are roughly 4,800,000 rubles away from the true house price on average. Furthermore, our linear regression model explains around 33.87% of the variance in house prices. 

We are only using this model for predictions and not interpreting the coefficients and so checking the model assumptions is not neccesary.